baybe/searchspace/continuous.py

```diff
@@ -273,6 +304,71 @@ def _drop_parameters(self, parameter_names: Collection[str]) -> SubspaceContinuo
             ],
         )
 
+    def _ensure_nonzero_parameters(
+        self,
+        inactive_parameters: Collection[str],
+        inactivity_threshold: float = 1e-5,
+    ) -> SubspaceContinuous:
+        """Create a new subspace with following several actions.
+
+        * Ensure active parameter != 0.0.
+        * Remove cardinality constraint.
+
+        Args:
+            inactive_parameters: A list of inactive parameters.
+            inactivity_threshold: Threshold for checking whether a value is zero.
+
+        Returns:
+            A new subspace object.
+        """
+        # TODO: Revise function name/docstring and arguments. In particular: why
+        #   does the function expect the inactive parameters instead of the active ones?
+
+        # TODO: Shouldn't the x != 0 constraints be applied on the level of the
+        #   individual constrains, also taking into account whether min_cardinality > 0?
+
+        # TODO: Instead of adding additional constraints, why not alter the parameter
+        #   bounds? In case we keep the constraints: is the sign of the threshold
+        #   correct?
```

Comment:
Here some points for discussion

Reply:
I made the following changes:

- Ensure active parameters != 0 by altering the parameter bounds
- By introducing `_FixedNumericalContinuousParameter` class, we ensure inactive parameters = 0 and active parameters != 0 in the copied subspace object. This is one way of communicating the subspace with inactive parameters. As discussed, we could also go with the option of the reduced subspace, then we need to manipulated the `_botorch_acqf`.

Reply:
closing this thread due to the larger refactoring I did. will open new threads at places where alignment is needed